source/sundials/kinsol.cc

```diff
@@ -524,6 +542,8 @@ namespace SUNDIALS
     reinit_vector = [](VectorType &) {
       AssertThrow(false, ExcFunctionNotProvided("reinit_vector"));
     };
+
+    setup_jacobian = [](const VectorType &, const VectorType &) { return 0; };
```

Comment:
Wouldn't you want this function to throw an assert, just like 3 lines higher up?

Reply:
Not really. We claim it is optional. Indeed the older version worked without this method. The new version, however, requires the setup. Providing a "do-nothing" setup by default should maintain backward compatibility and allow one not to specify any setup if required.

Reply:
I see. In that case, can you please move this line out of this function? The function is called `set_functions_to_trigger_an_assert()` :-)

I think the place where we do the general KINSOL set-up would be right.